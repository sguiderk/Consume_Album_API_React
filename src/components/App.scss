body {
	margin: 0;
	padding: 0;
	font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans',
		'Droid Sans', 'Helvetica Neue', sans-serif;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
}
.App {
	display: flex;
	align-items: center;
	justify-content: center;
	text-align: center;
	width: 100vw;
	color: #4e4e50;
	background-color: #1a1a1d;
	&-window {
		width: 95vw;
		background: #f4f7f6;
		border-radius: 5px;
		&-title {
			font-size: 3rem;
		}
	}
}
.container {
	width: 80%;
	margin: 2rem auto;
}
.gallery {
	display: flex;
	flex-wrap: wrap;
	padding: 0 4px;
}
.gallery__img {
	width: 100%;
	height: 100%;
	object-fit: cover;
	display: block;
}
.gallery__item {
	flex: 25%;
	max-width: 25%;
	position: relative;
	clear: both;
	padding-left: 35px;
	padding-bottom: 35px;
}
.gallery__info {
	background-color: #fcfcfc;
}
.gallery__figure {
	height: 80px;
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis;
	background: #dddddd;
	margin: 0;
	padding: 20px;
	position: relative;
}
.rollover {
	display: block;
	background: #E6EAEA;
	transition: all 0.3s ease-in-out;
}
.cover-rollover {
	width: 100%;
	height: auto;
	z-index: 1;
	transition: all 0.3s ease-in-out;
	-webkit-transform: translateZ(0);
	transform: translateZ(0);
}
.rollover img {
	height: auto;
	z-index: 1;
	transition: all 0.3s ease-in-out;
	-webkit-transform: translateZ(0);
	transform: translateZ(0);
}
.rollover:not(.no-hover):hover .cover-rollover img{
	opacity: 0.5;
}
.rollover:hover:not(.no-scale) .cover-rollover img{
	-webkit-transform: scale(1.05);
	transform: scale(1.05);
}
.rollover .open-image{
	position: absolute;
	top: 50px;
	left: 70px;
}
.rollover .hover-item {
	visibility: hidden;
	opacity: 0;
	z-index: 3;
	transition: opacity 0.3s ease-in-out;
}
.rollover:hover .hover-item {
	visibility: visible;
	opacity: 1;
}
.gallery__content {
	padding: 20px 20px 20px 20px;
}
.gallery__footer {
	position: relative;
	display: -webkit-box;
	display: flex;
	-webkit-box-pack: justify;
	justify-content: space-between;
	padding: 20px 20px 20px 20px;
}
.gallery__footer:before {
	content: '';
	position: absolute;
	top: 0;
	left: 20px;
	width: calc(100% - 80px);
	height: 1px;
	padding: 0 20px;
	background: #f4f7f6;
}
@media screen and (max-width: 800px) {
	.gallery__item {
		flex: 50%;
		max-width: 50%;
	}
}
@media screen and (max-width: 600px) {
	.gallery__item {
		flex: 100%;
		max-width: 100%;
	}
}
