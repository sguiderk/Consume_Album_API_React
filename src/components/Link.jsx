import React from 'react'
import { Link } from 'react-router-dom';
import './Link.scss';

export const AnchorLink = ({ children, link }) => (
	<a className="Link" target="_blank" rel="noopener noreferrer" href={link}>
		{children}
	</a>
);

export const AppLink = linkProps => <Link {...linkProps} className="Link"></Link>;