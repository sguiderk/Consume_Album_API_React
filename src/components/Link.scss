.Link {
	text-decoration: none;
	cursor: pointer;
	color: #c3073f;
}
