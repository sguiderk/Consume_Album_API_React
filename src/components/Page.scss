.Page {
	display: flex;
	flex-direction: column;
	&-title {
		margin-bottom: 2%;
		font-size: 1.5rem;
		display: flex;
		align-items: center;
		justify-content: center;
	}
	&-content {
		display: flex;
		align-items: center;
		justify-content: center;
		flex-direction: column;
		&-loading {
			animation: App-loading-spin infinite 1s linear;
			height: 50px;
			width: 50px;
			border: 5px solid #6f2232;
			border-top-color: #c3073f;
			border-radius: 50px;
		}
	}
}

@keyframes App-loading-spin {
	from {
		transform: rotate(0deg);
	}
	to {
		transform: rotate(360deg);
	}
}
