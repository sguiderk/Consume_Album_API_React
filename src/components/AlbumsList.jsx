import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { List, ListItem } from 'components/List';
import { AnchorLink, AppLink } from 'components/Link'
import { Page } from 'components/Page';
import { loadAlbums, loadUser } from 'redux/reducer';
import { TablePagination } from '@trendmicro/react-paginations';
import '@trendmicro/react-paginations/dist/react-paginations.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft ,faArrowRight, faUser } from '@fortawesome/free-solid-svg-icons';

const mapStateToProps = (state, ownProps) => ({
	loading: state.loadingAlbums,
	albums: state.albums,
	users: state.users,
});

const mapDispatchToProps = {
	initialLoad: loadAlbums,
	loadUser: loadUser,
};

export class AlbumsList extends Component {
	static propTypes = {
		initialLoad: PropTypes.func,
		loading: PropTypes.bool,
		albums: PropTypes.arrayOf(
			PropTypes.shape({
				id: PropTypes.number,
				title: PropTypes.string,
				year: PropTypes.number,
				user : PropTypes.object,
			})
		),
		users: PropTypes.arrayOf(
			PropTypes.shape({
				id: PropTypes.number,
				name: PropTypes.string,
				email: PropTypes.string,
			})
		),
	};
	state = {
		page: 1,
		pageLength: 10,
		totalRecords: 100
	};
	static defaultProps = {
		loading: true,
		albums: [],
		users:[]
	};
	componentDidMount() {
		this.props.initialLoad(1, 10);
		this.props.loadUser();
	}

	render() {

		const { loading, albums , users } = this.props;
		const state = { ...this.state };
		albums.map( album => (album.user = (users.find( x => x.id === album.userId)) ));

		return (
			<Page title="List of album" loading={loading}>
					<div className="container">
						<div className="gallery">
							{albums.map(album => (
								<div className="gallery__item"  key={album.id} >
									<figure className="gallery__figure rollover"  >
										<AppLink to={`/${album.id}`}>
											<div className="cover-rollover">
											{album.title}
											</div>
										</AppLink>
									</figure>
									<div className="gallery__info">
										<div className="gallery__content">
											<div className="row">
												<h5>{album.title}</h5>
											</div>
											<div className="row row-2col">
												<div className="box-left">From Croatia</div>
												<div className="box-right">
													April 09, 2019
												</div>
											</div>
										</div>
										<div className="gallery__footer">
											<div className="box-left">
												<div className="box-byuser js-tooltip-user">
													<div className="item js-user" data-username="MldGautier">
														<FontAwesomeIcon icon={ faUser }/>
													</div>
													<div className="by">
														<strong>By</strong>{album.user ?album.user.name:null}
													</div>
												</div>
											</div>
										</div>
									</div>
								</div>
							))}
						</div>
					</div>
				<TablePagination
					type="full"
					page={state.page}
					pageLength={state.pageLength}
					totalRecords={state.totalRecords}
					onPageChange={({ page, pageLength }) => {
						this.setState({ page, pageLength })
						this.props.initialLoad(page, pageLength);
					}}
					pageLengthMenu={[10,20,50]}
					prevPageRenderer={() => <FontAwesomeIcon icon={ faArrowLeft }/>}
					nextPageRenderer={() => <FontAwesomeIcon icon={ faArrowRight }/>}
				/>
			</Page>
		);
	}
}

export default connect(mapStateToProps,mapDispatchToProps)(AlbumsList);
