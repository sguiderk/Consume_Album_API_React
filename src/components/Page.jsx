import React from 'react';
import { AppLink } from 'components/Link';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {faArrowCircleLeft} from '@fortawesome/free-solid-svg-icons';

export const Page = ({ title, loading, backButton, children }) => (
	<div className="Page">
		<header className="Page-title">{backButton && <><AppLink to="/"><FontAwesomeIcon icon={ faArrowCircleLeft }/></AppLink></>}{title}</header>
		<div className="Page-content">{loading ? <div className="Page-content-loading" /> : children}</div>
	</div>
);
