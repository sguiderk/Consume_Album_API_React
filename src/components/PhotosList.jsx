import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { loadPhotos,loadUser } from 'redux/reducer';
import { Page } from 'components/Page';
import { TablePagination } from '@trendmicro/react-paginations';
import '@trendmicro/react-paginations/dist/react-paginations.css';
import Modal from 'react-modal';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {faArrowLeft, faArrowRight, faSearch, faTimesCircle, faUser} from '@fortawesome/free-solid-svg-icons';


const mapStateToProps = (state, ownProps) => ({
	loading: state.loadingPhotos,
	photos: state.photoAlbums[ownProps.idAlbum] || [],
	users: state.users,
	album: state.albums.find(album => album.id === parseInt(ownProps.idAlbum, 10)),
});

const mapDispatchToProps = {
	loadPhotos: loadPhotos,
	loadUser: loadUser,
};

const customStyles = {
	content : {
		top                   : '50%',
		left                  : '50%',
		right                 : 'auto',
		bottom                : 'auto',
		marginRight           : '-50%',
		transform             : 'translate(-50%, -50%)'
	}
};

export class PhotosList extends Component {

	constructor() {
		super();
		this.openModal = this.openModal.bind(this);
		this.afterOpenModal = this.afterOpenModal.bind(this);
		this.closeModal = this.closeModal.bind(this);
	}

	openModal(id) {
		this.setState({modalIsOpen:
				{
					[id]: true
				}
		});
	}

	afterOpenModal() {
		this.subtitle.style.color = '#000';
	}

	closeModal() {
		this.setState({modalIsOpen: false});
	}

	static propTypes = {
		loading: PropTypes.bool,
		photos: PropTypes.arrayOf(
			PropTypes.shape({
				albumId: PropTypes.number,
				id: PropTypes.number,
				title: PropTypes.string,
				url: PropTypes.string,
			})
		),
		album: PropTypes.shape({
			id: PropTypes.number,
			title: PropTypes.string,
			idAlbum: PropTypes.number,
			user : PropTypes.object,
		}),
		users: PropTypes.arrayOf(
			PropTypes.shape({
				id: PropTypes.number,
				name: PropTypes.string,
				email: PropTypes.string,
			})
		),
		idAlbum: PropTypes.string,
		loadPhotos: PropTypes.func,
	};

	state = {
		page: 1,
		pageLength: 10,
		totalRecords: 50,
		modalIsOpen: false,
	};

	componentDidMount() {
		const { photos, loadPhotos, idAlbum } = this.props;
		this.props.loadUser();
		photos.length === 0 && loadPhotos(idAlbum,1,10);
	}

	render() {
		const { photos, idAlbum, loading, album ,users } = this.props;
		const state = { ...this.state };



		return (
			<Page title={`${album?album.title:''}`} loading={loading} backButton>
				<div className="by">
					<div className="item js-user" data-username="MldGautier">
						<FontAwesomeIcon icon={ faUser }/>
					</div>
					<strong>By</strong>{album?album.user.name:''}
				</div>
				<div className="container">
					<div className="gallery">
					{photos.map(photo => {
						return (
							<div className="gallery__item" key={photo.id} >
								<figure className="gallery__figure rollover">
									<div className="cover-rollover">
									<img src={photo.url} width='100px' />
									</div>
									<div className="hover-item ">
										<FontAwesomeIcon className="open-image" onClick= {this.openModal.bind(this, photo.id)} icon={ faSearch }/>
									</div>
								</figure>
								{ photo.title }
								<Modal
									isOpen={this.state.modalIsOpen[photo.id]}
									onAfterOpen={this.afterOpenModal}
									onRequestClose={this.closeModal}
									style={customStyles}
									contentLabel="Example Modal"
								>
									<FontAwesomeIcon onClick={this.closeModal} icon={ faTimesCircle }/>
									<h4 ref={subtitle => this.subtitle = subtitle}>{photo.title}</h4>
									<div className="by">
										<div className="item js-user" data-username="MldGautier">
											<FontAwesomeIcon icon={ faUser }/>
										</div>
										<strong>By</strong>{album?album.user.name:''}
									</div>
									<img src={photo.url}  />
								</Modal>
							</div>
						);
					})}
					</div>
				</div>
				<TablePagination
					type="full"
					page={state.page}
					pageLength={state.pageLength}
					totalRecords={state.totalRecords}
					onPageChange={({ page, pageLength }) => {
						this.setState({ page, pageLength })
						this.props.loadPhotos(this.props.idAlbum, page, pageLength);
					}}
					prevPageRenderer={() => <FontAwesomeIcon icon={ faArrowLeft }/>}
					nextPageRenderer={() => <FontAwesomeIcon icon={ faArrowRight }/>}
				/>
			</Page>
		);
	}
}

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(PhotosList);
