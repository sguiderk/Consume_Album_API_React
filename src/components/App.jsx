import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { loadAlbums } from 'redux/reducer';
import Header from 'components/Header';
import Footer from 'components//Footer';
import AlbumsList from 'components/AlbumsList';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import PhotosList from 'components/PhotosList';

const mapDispatchToProps = {
	initialLoad: loadAlbums,
};

class App extends Component {
	constructor(props) {
		super(props);
		this.state = {
			header: "Consume Album API",
			footer: "Content Footer",
		}
	}
	static propTypes = {
		initialLoad: PropTypes.func,
	};
	render() {
		return (
			<Router>
				<div className="App">
					<div className="App-window">
						<Header headerProp = {this.state.header}/>
						<h1 className="App-window-title">Album App</h1>
						<Route path="/" exact render={() => <AlbumsList />} />
						<Route path="/:idAlbum" render={({ match }) => <PhotosList idAlbum={match.params.idAlbum} />} />
						<Footer footerProp = {this.state.footer}/>
					</div>
				</div>
			</Router>
		);
	}
}

export default connect(
	null,
	mapDispatchToProps
)(App);
