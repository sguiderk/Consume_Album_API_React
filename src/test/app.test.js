configure({ adapter: new Adapter() });
import Enzyme, { shallow, render, mount } from 'enzyme';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App  from '../components/App';
import AlbumsList from '../components/AlbumsList';
import PhotosList from '../components/PhotosList';
import { AnchorLink } from '../components/Link'
import { Page } from '../components/Page';
import store from 'redux/storeConfig';
import { Provider } from 'react-redux';


describe('creaction of component AlbumsList,PhotosList,List,AnchorLink and Page', () => {

    it('component App created', () => {
        const renderedComponent = shallow(<App />);
        expect(renderedComponent).toMatchSnapshot();
    });
    it('component AlbumsList created', () => {
        const renderedComponent = shallow(<AlbumsList />);
        expect(renderedComponent).toMatchSnapshot();
    });
    it('component PhotosList created', () => {
        const renderedComponent = shallow(<PhotosList />);
        expect(renderedComponent).toMatchSnapshot();
    });
    it('component AnchorLink created', () => {
        const renderedComponent = shallow(<AnchorLink />);
        expect(renderedComponent).toMatchSnapshot();
    });
    it('component Page created', () => {
        const renderedComponent = shallow(<Page />);
        expect(renderedComponent).toMatchSnapshot();
    });


})


describe('App contains AlbumsList,PhotosList,List,AnchorLink and Page', () => {
    it('App contains Header', function() {
        expect(shallow(<App />).contains(<AlbumsList/>)).toMatchSnapshot();
    });
    it('App contains PhotosList', function() {
        expect(shallow(<App />).contains(<PhotosList/>)).toMatchSnapshot();
    });
    it('App contains PhotosList', function() {
        expect(shallow(<App />).contains(<PhotosList/>)).toMatchSnapshot();
    });
    it('App contains Page', function() {
        expect(shallow(<App />).contains(<Page/>)).toMatchSnapshot();
    });
    it('App contains AnchorLink', function() {
        expect(shallow(<App />).contains(<AnchorLink/>)).toMatchSnapshot();
    });
})

describe('Has Title created Album App', () => {

    it('Title created', () => {
        const component = mount(<Provider store={store}>
            <App/>
        </Provider>);

        expect(component.find("h1").text()).toEqual("Album App");
    });

})


describe('Has header List of album', () => {

    it('header created', () => {
        const component = mount(<Provider store={store}>
            <AlbumsList/>
        </Provider>);

        expect(component.find("header").text()).toEqual("List of album");
    });

})

describe('Check state of AlbumsList is empty', () => {

    it('state of AlbumsList is empty', () => {

        const cmp = shallow(<Provider store={store}>
            <AlbumsList />
        </Provider>);

        expect(cmp.state().storeState.albums).toEqual([]);

    });

})


describe('Check state of PhotosList is empty', () => {

    it('state of PhotosList is empty', () => {

        const cmp = mount(<Provider store={store}>
            <AlbumsList />
        </Provider>);

        expect(cmp.state().storeState.photos).toEqual(undefined);
    });

})
