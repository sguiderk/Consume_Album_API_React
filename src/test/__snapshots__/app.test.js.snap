// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App contains AlbumsList,PhotosList,List,AnchorLink and Page App contains AnchorLink 1`] = `false`;

exports[`App contains AlbumsList,PhotosList,List,AnchorLink and Page App contains Header 1`] = `false`;

exports[`App contains AlbumsList,PhotosList,List,AnchorLink and Page App contains List 1`] = `true`;

exports[`App contains AlbumsList,PhotosList,List,AnchorLink and Page App contains Page 1`] = `false`;

exports[`App contains AlbumsList,PhotosList,List,AnchorLink and Page App contains PhotosList 1`] = `false`;

exports[`App contains AlbumsList,PhotosList,List,AnchorLink and Page App contains PhotosList 2`] = `false`;

exports[`creaction of component AlbumsList,PhotosList,List,AnchorLink and Page component AlbumsList created 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(AlbumsList) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`creaction of component AlbumsList,PhotosList,List,AnchorLink and Page component AnchorLink created 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AnchorLink />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Link",
      "href": undefined,
      "rel": "noopener noreferrer",
      "target": "_blank",
    },
    "ref": null,
    "rendered": null,
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Link",
        "href": undefined,
        "rel": "noopener noreferrer",
        "target": "_blank",
      },
      "ref": null,
      "rendered": null,
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`creaction of component AlbumsList,PhotosList,List,AnchorLink and Page component App created 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(App) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`creaction of component AlbumsList,PhotosList,List,AnchorLink and Page component Page created 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="Page-title"
        />,
        <div
          className="Page-content"
        />,
      ],
      "className": "Page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
          ],
          "className": "Page-title",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "Page-content",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="Page-title"
          />,
          <div
            className="Page-content"
          />,
        ],
        "className": "Page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
            ],
            "className": "Page-title",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "Page-content",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`creaction of component AlbumsList,PhotosList,List,AnchorLink and Page component PhotosList created 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(PhotosList) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
